app/client/components/FormRenderer.ts

```diff
@@ -419,8 +420,7 @@ class ChoiceRenderer extends BaseFieldRenderer  {
           choices.reverse();
         }
       }
-      // Support for 1000 choices. TODO: make limit dynamic.
-      this._choices = choices.slice(0, 1000);
+      this._choices = choices.slice();
```

Comment:
We may even not copy the choices and reuse the same array:

```suggestion
      this._choices = choices
```

BTW, any reason for originally limiting to 1000 items? (for performance maybe?)

Reply:
My guess is yes, it was for performance reasons, because when rendering all options, too many cause browser to get slow. As for .slice(), I agree it's better remove it if it isn't needed. In line 779 too. If you find that it is needed after all, then a comment would be nice about why it matters.